diff --git a/RegExp.java b/RegExp.java
index 1290c85..e098704 100644
--- a/RegExp.java
+++ b/RegExp.java
@@ -492,6 +492,10 @@ public class RegExp {
 		r.digits = digits;
 		return r;
 	}
+	
+	private static RegExp makeNewline() {
+		return makeUnion(makeUnion(makeString("\r"), makeString("\n")), makeString("\r\n"));
+	}
 
 	private boolean peek(String s) {
 		return more() && s.indexOf(b.charAt(pos)) != -1;
@@ -619,7 +623,9 @@ public class RegExp {
 	}
 
 	private final RegExp parseSimpleExp() throws IllegalArgumentException {
-		if (match('.'))
+		if (match('$'))
+			return makeNewline();
+		else if (match('.'))
 			return makeAnyChar();
 		else if (check(EMPTY) && match('#'))
 			return makeEmpty();